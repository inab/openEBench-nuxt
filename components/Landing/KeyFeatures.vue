<template>
	<v-container>
		<h2 class="text-h4 mb-15">What does OpenEBench offer?</h2>
		<v-row column>
			<v-col v-if="$vuetify.breakpoint.mdAndUp" md="6">
				<v-img :src="illustration" contain max-height="350" />
			</v-col>
			<v-col
				class="d-flex flex-column justify-center"
				sm="12"
				offset-md="0"
				md="6"
			>
				<div class="text-h5 d-flex align-center">
					<v-icon color="accent" class="mr-3" x-large>mdi-check-circle</v-icon>
					Supporting community-driven benchmarking
				</div>
				<div class="text-h5 d-flex align-center mt-10">
					<v-icon color="accent" class="mr-3" x-large>mdi-check-circle</v-icon
					>Monitoring of software quality in life sciences
				</div>
				<div class="text-h5 d-flex align-center mt-10">
					<v-icon color="accent" class="mr-3" x-large>mdi-check-circle</v-icon
					>Contributing towards the adoption of best practices for research
					software development
				</div>
			</v-col>
		</v-row>
	</v-container>
</template>

<script>
export default {
	name: 'WelcomeSection',
	data() {
		return {
			illustration: require('~/static/images/illustrations/404.svg'),
		};
	},
};
</script>
