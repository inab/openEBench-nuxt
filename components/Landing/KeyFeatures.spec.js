import { mount } from '@vue/test-utils';
import KeyFeatures from './KeyFeatures.vue';

const factory = () => {
	return mount(KeyFeatures, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
	});
};

describe('KeyFeatures', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});
});
