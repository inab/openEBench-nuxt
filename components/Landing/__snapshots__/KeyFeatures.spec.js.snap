// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyFeatures should match snapshot 1`] = `
"<div class=\\"container\\">
  <h2 class=\\"text-h4 mb-15\\">What does OpenEBench offer?</h2>
  <div column=\\"\\" class=\\"row\\">
    <div class=\\"col-md-6 col\\">
      <div class=\\"v-image v-responsive theme--light\\" style=\\"max-height: 350px;\\">
        <div class=\\"v-responsive__content\\"></div>
      </div>
    </div>
    <div class=\\"d-flex flex-column justify-center col-sm-12 col-md-6 offset-md-0 col\\">
      <div class=\\"text-h5 d-flex align-center\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mr-3 mdi mdi-check-circle theme--light accent--text\\" style=\\"font-size: 40px;\\"></i>
        Supporting community-driven benchmarking
      </div>
      <div class=\\"text-h5 d-flex align-center mt-10\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mr-3 mdi mdi-check-circle theme--light accent--text\\" style=\\"font-size: 40px;\\"></i>Monitoring of software quality in life sciences
      </div>
      <div class=\\"text-h5 d-flex align-center mt-10\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mr-3 mdi mdi-check-circle theme--light accent--text\\" style=\\"font-size: 40px;\\"></i>Contributing towards the adoption of best practices for research
        software development
      </div>
    </div>
  </div>
</div>"
`;
